src/core/js/models/configModel.js

```diff
@@ -16,13 +16,36 @@ export default class ConfigModel extends LockingModel {
     };
   }
 
+  /**
+   * Parse the incoming search queries for a 'lang' parameter value to return
+   */
+  getLanguageFromURL() {
+    const search = window.location.search.substring(1);
+    const queries = search.split('&');
+
+    queries.forEach((query) => {
+      const pair = query.split('=');
+
+      if (decodeURIComponent(pair[0]) !== 'lang') return;
+
+      return decodeURIComponent(pair[1]);
+    });
+  }
+
   initialize(attrs, options) {
     this.url = options.url;
     // Fetch data & if successful trigger event to enable plugins to stop course files loading
     // Then check if course files can load
     // 'configModel:loadCourseData' event starts the core content collections and models being fetched
     this.fetch({
       success: () => {
+        const language = this.getLanguageFromURL();
+
+        if (language) {
+          this.set('_defaultLanguage', language);
+          this.set('_activeLanguage', language);
```

Comment:
are you sure this line is required? thought it would handle this automatically elsewhere